### ISSUE TITLE
Move the regex declaration outside the function

### ISSUE THREAD

**Issue by morellodev** [sindresorhus/is-absolute-url#15]
Hi @sindresorhus, this might be a very little performance improvement, but it makes the code even clearer. Instead of inlining the regex for testing absolute urls inside the body of the function, it would be better to extract it outside and then referencing it:

```javascript
const ABSOLUTE_URL_REGEX = /^[a-zA-Z][a-zA-Z\d+\-.]*?:/;

export default function isAbsoluteUrl(url) {
  ...
  return ABSOLUTE_URL_REGEX.test(url);
}
```

**Comment by sindresorhus** [comment]
I doubt it makes a difference performance-wise, but I agree it's cleaner this way.

**Comment by morellodev** [comment]
Yes this does not make great difference in terms of performance, but now those regexes are created once and reused every time the function is called.
